daemon/containerd/image.go

```diff
@@ -278,9 +279,10 @@ func (i *ImageService) resolveImage(ctx context.Context, refOrID string) (contai
 
 	// If the identifier could be a short ID, attempt to match
 	if truncatedID.MatchString(refOrID) {
+		idWithoutAlgo := strings.TrimPrefix(refOrID, "sha256:")
```

Comment:
Do you know if the non-containerd variant actually allows ID prefixes here? The `stringed. IsShortID` function only accepts short (fixed length, 12-characters) ID; https://github.com/moby/moby/blob/152036f0aab931439c4e18647335956b071edd01/pkg/stringid/stringid.go#L18-L31


(Also slightly wondering if trimming the algorithm before matching it would be an alternative (I haven't benchmarked the regex though))

Reply:
It does, there is a test for it

Reply:
Thanks! LOL, so much for consistency 😂 

wouldn't surprise me if the test was written based on "it works" and not "what was meant to work"

Reply:
Did a quick check; this was the PR adding that test;

- https://github.com/moby/moby/pull/18409
- https://github.com/moby/moby/pull/18443
- related to https://github.com/distribution/distribution/pull/1231

Funny thing is that that last one seems to be changing things to use _longer_ digests in tests (and removing the short ones).


Oh, wait, but that test _does_ use `12` characters (originally it took `10`, but updated in the second PR above)? Did it overlook the `sha256:` prefix in that length 🤔 ? https://github.com/moby/moby/blob/6dcefa3739b2aa3eb7cb3a7b6ed4faa4eae33fdc/integration-cli/docker_cli_inspect_test.go#L308-L317